dsref/load.go

```diff
@@ -0,0 +1,47 @@
+package dsref
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/qri-io/dataset"
+	qerr "github.com/qri-io/qri/errors"
+)
+
+// Loader loads and opens a dataset. The only useful implementation of the
+// loader interface is in github.com/qri-io/qri/lib.
+// TODO(b5) - This interface is a work-in-progress
+type Loader interface {
+	LoadDataset(ctx context.Context, ref Ref, source string) (*dataset.Dataset, error)
+}
+
+// ParseResolveLoad is the function type returned by NewParseResolveLoader
+type ParseResolveLoad func(ctx context.Context, refStr string) (*dataset.Dataset, error)
+
+// NewParseResolveLoadFunc composes a username, resolver, and loader into a
+// higher-order function that converts strings to full datasets
+// pass the empty string as a username to disable the "me" keyword in references
+func NewParseResolveLoadFunc(username string, resolver Resolver, loader Loader) ParseResolveLoad {
```

Comment:
This implementation shouldn't be in this package. The purpose of `dsref` is that it's a low-dependency foundation that other things can build on, and so it shouldn't have any opinionated decisions or magic behavior in it. Even if this just looks like concatenating existing interfaces, the reliance of having a valid username, the "me" convention, and user-centric error messages are all business logic that should live higher up. As the comment above implies, lib has the useful implementation of the loader, so it should be constructing this function, and passing it down to the subsystems that need it.

Reply:
v good points. moving to lib